app/ts/simulation/compoundGovernanceFaking.ts

```diff
@@ -41,32 +45,32 @@ export const simulateCompoundGovernanceExecution = async (ethereumClientService:
 		}
 	]
 	const governanceContractCalls = await ethereumClientService.multicall(calls, [], parentBlock.number)
+	governanceContractCalls.forEach((call) => { if (call.statusCode !== 'success') throw new Error('Failed to retrieve governance contracts information') })
 	if (governanceContractCalls[0]?.returnValue === undefined) return undefined
 	const timeLockContractResult = compoundGovernanceAbi.decodeFunctionResult('timelock', governanceContractCalls[0].returnValue)
 	const timeLockContract = EthereumAddress.parse(timeLockContractResult[0])
 	if (governanceContractCalls[1] === undefined) throw new Error('proposals return value was undefined')
-	const [_id, _proposer, eta, _startBlock, _endBlock, _forVotes, _againstVotes, _abstainVotes, canceled, executed] = compoundGovernanceAbi.decodeFunctionResult('proposals', governanceContractCalls[1].returnValue)
+	const proposal = compoundGovernanceAbi.decodeFunctionResult('proposals', governanceContractCalls[1].returnValue)
+	const eta: bigint = funtypes.BigInt.parse(proposal[0].eta)
```

Comment:
maybe `safeParse` here so it doesn't throw for whatever unknown reason

Reply:
I want it to throw in this case and stop the execution as that means the contract behaves differently than I assume, and that probably means I am doing something wrong. We should not show simulation results if the contract behaves differently than I assume, as I would perform wrong simulation